### DISCUSSION TITLE
What are the available values to use in the training program xml?

### DISCUSSION THREAD

**Discussion by ndesai510** [cagnulein/qdomyos-zwift#1149] [General]
For example,
https://github.com/cagnulein/qdomyos-zwift/blob/master/train-programs-examples/echelon.xml
duration and resistance are in that xml.

What are all the various values that can be used?

I am trying to add a target cadence value and based on your Training Program Saver (http://80.211.67.253:8081/training.htm), cadence should work, however, it doesnt populate the "T.Cadence" tile when the program is loaded.

Wonder if I am just adding it incorrectly, but it would be great to know all the available options that can be used in the xml files.

Thanks!

**Comment by cagnulein** [comment]
hi you can find here https://github.com/cagnulein/qdomyos-zwift/blob/master/src/trainprogram.cpp from line 900
if you need other variables i can add easily!
let me know

**Reply by ndesai510** [reply]
Thanks for the reference.
It seems as though "cadence" does not work in the Target Cadence tile.

For example, I set cadence to 18, upper_cadence to 19 and lower_cadence to 17 in the xml.  The cadence stays at 0 no matter what.  I can see the upper and lower cadence in the tile, but the main cadence value is always 0.

I am using a rower and have set the virtual rower option.

Please let me know if you can replicate. I can add logs tomorrow if needed.

Thanks!

**Reply by cagnulein** [reply]
ok i will check this. thanks

**Reply by cagnulein** [reply]
fixed! Do you need the beta or are you on a raspberry? @ndesai510

**Reply by ndesai510** [reply]
Awesome. I'm currently on Android 2.12.43. Will the next version have the update?

Thanks for your continued support with this great project!

**Reply by cagnulein** [reply]
yes i will build this tomorrow!

**Reply by cagnulein** [reply]
uploaded!

**Reply by nieko-nera** [reply]
> hi you can find here https://github.com/cagnulein/qdomyos-zwift/blob/master/src/trainprogram.cpp from line 900 if you need other variables i can add easily! let me know

can you add HR min (minimum Heart rate) and HR max (maximum heart rate)?

**Comment by cagnulein** [comment]
how should i use these?

Il giorno dom 12 feb 2023 alle 14:53 Tadas ***@***.***> ha
scritto:

> hi you can find here
> https://github.com/cagnulein/qdomyos-zwift/blob/master/src/trainprogram.cpp
> from line 900 if you need other variables i can add easily! let me know
>
> can you add HR min (minimum Heart rate) and HR max (maximum heart rate)?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cagnulein/qdomyos-zwift/discussions/1149#discussioncomment-4949327>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAALYWDINR4NKFETD7CQGUTWXDTNJANCNFSM6AAAAAAT2VWE6I>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>
-- 
Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud

**Reply by nieko-nera** [reply]
same as HR zones, but you define range yourself (so you don't need to calculate what HR zone 2 is), for example maintain HR between 120 and 130. And if possible add this to "Training Program Options" in QZ settings

**Reply by cagnulein** [reply]
mmm @nieko-nera it seems a little redundant to me. I mean your typical usage will be to restrict a zone or to a enlarge a zone?

**Comment by nieko-nera** [comment]
If i use HR zones i dont see what is actual HR range is (how its calculated). For example today i feel little tired, i can lover HR range by 5 bpm.

**Reply by cagnulein** [reply]
ok let's convert this to a new ticket
